academic-services/knowledge-exploration-service/tutorial-grammar-design.md

```diff
@@ -0,0 +1,392 @@
+---
+title: Add search to the library browser
+description: Step by step tutorial to design MAKES grammar for custom data
+ms.topic: tutorial
+ms.date: 11/16/2020
+---
+# Add search to the library browser
+
+This tutorial is a continuation of the ["Build a library browser with contextual filters"](tutorial-entity-linking.md) tutorial.
+
+This tutorial illustrates how to
+
+- Create a MAKES Index tailored for library publication entities search
+- Design a MAKES grammar to process natural language search queries
+- Build, compile and deploy a MAKES instance with custom index and grammar
+- Extend the library browser with search capability using MAKES APIs
+
+![Library search application](media/privateLibraryExampleApp-search-homepage.png)
+
+## Prerequisites
+
+- [Microsoft Academic Knowledge Exploration Service (MAKES) subscription](get-started-setup-provisioning.md) (release version after 2020-11-23)
+- Completion of ["Link private publication records with MAKES entities"](tutorial-entity-linking.md) tutorial
+- Completion of ["Build a library browser with contextual filters"](tutorial-schema-design.md) tutorial
+- Read through ["How to define index schema"](how-to-index-schema.md) how-to guide
+- Read through ["How to define grammar"](how-to-grammar.md) how-to guide
+- Download the [sample search schema for linked private library publications](samplePrivateLibraryData.linked.search.schema.json)
+- Download the [sample search grammar for linked private library publications](samplePrivateLibraryData.linked.search.grammar.xml)
+
+## Create a searchable index
+
+In order to search the index, we will have modify the schema from ["Build a library browser with contextual filters"](tutorial-entity-linking.md) tutorial and add index operations on more attributes.
+
+### Modify schema to support search
+
+To make the index searchable, we extend the schema from the ["Build a library browser with contextual filters"](tutorial-schema-design.md) tutorial and add the following attributes and index operations.
```

Comment:
>and add the following attributes and index operations [](start = 148, length = 53)

These aren't being added, they're being modified. Can yo change the language to reflect that?

Also, it would be useful to have the table below reflect the original and modified values, i.e. a column that is "original index operations" and a column that is "new index operations" or something like that.

Reply:
All attribute shown here are new except the "composite" definitions.

Reply:
will remove the composite attributes from this table since it's already defined in previous tutorial

Reply:
All attribute shown here are new except the "composite" definitions. I will remove the composite attributes from this table since it's already defined in previous tutorial

---
In reply to: [533870893](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/194#discussion_r533870893) [](ancestors = 533870893)